modules/api/src/main/java/org/eclipse/xpanse/api/controllers/ServiceDeployerApi.java

```diff
@@ -199,13 +204,29 @@ public Response purge(@PathVariable("id") String id) {
     public UUID migrate(@Valid @RequestBody MigrateRequest migrateRequest) {
         UUID newId = UUID.randomUUID();
         Map<String, Object> variable = new HashMap<>();
-        variable.put("id", migrateRequest.getId());
-        variable.put("newId", newId);
-        variable.put("migrateRequest", migrateRequest);
-        workflowProcessUtils.asyncStartProcess("migrate", variable);
+        variable.put(MigrateConstants.ID, migrateRequest.getId());
+        variable.put(MigrateConstants.NEW_ID, newId);
+        variable.put(MigrateConstants.MIGRATE_REQUEST, migrateRequest);
+        variable.put(MigrateConstants.USER_ID, deployService.getCurrentLoginUserId());
+        workflowProcessUtils.asyncStartProcess(MigrateConstants.PROCESS_KEY, variable);
         return newId;
     }
 
+    /**
+     * Get the status of service migration by service ID.
+     *
+     * @param id service ID.
+     */
+    @Tag(name = "Service", description = "APIs to manage the service instances")
+    @Operation(description = "Get the status of service migration by service ID")
+    @GetMapping(value = "/services/migration/state/{id}", produces =
+            MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public Response getMigrateServiceStateById(@PathVariable("id") String id) {
```

Comment:
Even here it is better to define a new data model for the state. Even if it has just one field, it is ok.